### PR TITLE
added source "Ortofotomozaika SR"

### PR DESCRIPTION
**Pull request by zdila**

### CODE REVIEW

**Comment by zdila** [comment]
Is there anything from my side to help with merging this pull request? It's been 23 days already without any reaction.

**Comment by andrewharvey** [comment]
I'm just not sure about the license terms if that indeed allows use in OSM, has this been discussed anywhere else already?

**Comment by zdila** [comment]
@andrewharvey Thanks for your reply. License of the source data is at https://www.geoportal.sk/files/zbgis/objednavky/podmienky_poskytovania_a_pouzivania_udajov_ortofotomozaiky.pdf. Generally it says that we can use it for commercial or noncommercial purposes without any warranty and we must attribute the source with text "© GKÚ, NLC; r.2017 - 2019" - which is mentioned at https://wiki.openstreetmap.org/wiki/Contributors.

**Comment by zdila** [comment]
Hi @andrewharvey. Is there still something missing for merging this pull request?

**Comment by zdila** [comment]
Any chance for this to be merged? People are using old Bing maps and destroying what has alreay been mapped by more precise and actual sources. Is there anybody else I can discuss with to merge it? @grischard? @rbuffat? Anybody?

**Comment by andrewharvey** [comment]
> @andrewharvey Thanks for your reply. License of the source data is at https://www.geoportal.sk/files/zbgis/objednavky/podmienky_poskytovania_a_pouzivania_udajov_ortofotomozaiky.pdf. Generally it says that we can use it for commercial or noncommercial purposes without any warranty and we must attribute the source with text "© GKÚ, NLC; r.2017 - 2019" - which is mentioned at https://wiki.openstreetmap.org/wiki/Contributors.

That's probably okay, but I'm just not comfortable making that call without maybe the LWG giving it a stamp of approval.

**Comment by zdila** [comment]
@andrewharvey so what should we do here for LWG to give it stamp of approval?

**Comment by andrewharvey** [comment]
@simonpoole would you have time to review this license? I know with CC we need to ask for confirmation that attribution on the wiki page is enough as we can't attribute in every place OSM is used, so that's the part I was unsure about with this license.

Then there is also a section

> 3. May not further reproduce and distribute the source data in its original / supplied / form through any medium.

which doesn't sound good, but maybe it's just a bad translation.

**Comment by tjamecny** [comment]
> Then there is also a section
> 
> > 3. May not further reproduce and distribute the source data in its original / supplied / form through any medium.
> 
> which doesn't sound good, but maybe it's just a bad translation.

The translation is correct, but this condition applies to the **source** data, that we have obtained from the GKÚ. The source data consists from 3 regions (west / middle / east Slovakia). Martin joined the source data to one big geotiff, adjusted projection (S-JTSK), and generated the tiles. These tiles are provided from https://ofmozaika.tiles.freemap.sk/ (which is the subject of this pull request), and the **source** data are kept in a safe place, therefore we are NOT violating this condition.

**Comment by zdila** [comment]
Hello. A week has passed without a reply so let me kindly remind you.

**Comment by andrewharvey** [comment]
Reviewing the license translation at https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Sources#Ortofotomozaika_SR_.28orthophoto_by_GK.C3.9A_and_NLC.29 my interpretation and thoughts on suitability for use in OSM are below.

So I'm still not comfortable with using this data in OSM unless it has been discussed and it's clear we can use it in OSM.

> Source data is provided to the subscriber free of charge.

No issue, actually this part is irrelevant for use in OSM whether the data is available free or paid makes no difference to the license that data is made available under.

> The data subscriber is entitled to combine source data with other data, to use them in creating their own work for commercial and non-commercial purposes, create new work, publish work on the Internet, publish source data as the underlying map, publish source data in conjunction with other data / thematic layers through web applications.

What if someone took some source data included it in OSM, but then someone took OSM and filtered out only data from this source? Would that mean then it's not a new work and hence doesn't have the license grant? I think to include in OSM we can't have any lingering limitations for this scenario.

> The subscriber is obliged to state the author of the source data when creating his own work and when publishing it in the form "© GKÚ, NLC; r.2017. "

It's not clear exactly how such attribution can be made. People using OSM data are only expected to attribution OSM and not every source that OSM includes. OSM can provide attribution to contributors on the wiki page, but downstream users of OSM don't. This is one of the things we use the CC BY waiver to clarify, I'm not sure if the simple statement on attribution in this license is an issue or not.

> Copyrights enjoy protection under the relevant provisions of Act no. 185/2015 - Copyright Act, as amended, as well as the Criminal Code.Abuse of data by users may therefore have civil or criminal consequences.

**Comment by andrewharvey** [comment]
Is this source aerial imagery or a map?

**Comment by zdila** [comment]
It is aerial imagery.

**Comment by andrewharvey** [comment]
> It is aerial imagery.

I guess in that case any tracing is considered creating your own work derivative from the source, and so you can not go back and re-created the source exactly from the traced data. So that second point is probably fine then.

**Comment by simonpoole** [comment]
This is something that to my knowledge does not have a well established body of case law (anywhere). 

However as a rule, OSM will not violate ToU  or in general take advantage of grey areas of the application of copyright legislation. In other words: when in doubt, we need explicit permission.

Simon
-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit Kaiten Mail gesendet.

**Comment by zdila** [comment]
@simonpoole We have finally obtained explicit permission to use Ortofotomozaik SR from the authority.

Content of the email:

> From: Leitmannová Katarína, Ing. <Katarina.Leitmannova@skgeodesy.sk>
> Date: Tue, Oct 13, 2020 at 9:09 AM
> Subject: FW: Použitie Ortofotomozaiky SR pre účely mapovania do OpenStreetMap
> To: martin.zdila@freemap.sk <martin.zdila@freemap.sk>
> Cc: Michalík Ľuboslav <Luboslav.Michalik@skgeodesy.>, Mikova Anna (anna.mikova@nlcsk.org) <anna.mikova@nlcsk.org>
> 
> Dobrý deň,
> 
> týmto vyjadrujeme súhlas s využívaním ortofotomozaiky SR zo stránky https://www.geoportal.sk/sk/ na účely mapovania do OSM. Zároveň Vás upozorňujeme, že je potrebné dodržiavať licenčné podmienky - Podmienky poskytovania a používania údajov ortofotomozaiky.
> 
> Na stránke prispievateľov do OSM prosím upravte zdroj údajov takto:
> Data with source=Ortofotomozaika SR derived from National forestry center (Národné lesnícke centrum vo Zvolene) and Geodetic and Cartographic Institute Bratislava (Geodetický a kartografický ústav Bratislava) imagery. © GKÚ, NLC; r.2017-2019.
> 
> 
> S pozdravom
> 
> Ing. Katarína Leitmannová
> riaditeľka odboru geodézie a medzinárodných vzťahov
> Úrad geodézie, kartografie a katastra SR
> Chlumeckého 2, 820 12  Bratislava 212
> Tel.: +421 2 2081 6080
> e-mail: ugkkogmv@skgeodesy.sk 
> http://www.geodesy.gov.sk

I have already fulfilled their request to update tagging (source=Ortofotomozaika SR) and authority name on wiki.osm.org pages.

**Comment by zdila** [comment]
BTW how will iD (and other editors using this project) know that it should add source="Ortofotomozaika SR" changeset tag?

**Comment by rbuffat** [comment]
@zdila  I don't think that there is a mechanism that automatically adds a source tag:

See these discussions:
ID: https://github.com/openstreetmap/iD/issues/6300
JSOM: https://josm.openstreetmap.de/ticket/19911

**Comment by andrewharvey** [comment]
I don't believe we currently have any sources used in OSM that require object level attribution via tags because this is not possible to ensure happens. Most sources we just attribute on the Contributors wiki page, but basically make no guarantees you could even identify which changesets or objects used the source.

As new contributors come along and edit the features and they then remove the source tag, does that put us in violation of the attribution? I think that's a very risky position to put the project in where certain tags are legally mandated.

**Comment by zdila** [comment]
@andrewharvey  They actually only adjusted existing text we provided them from wiki. It is a simmilar thing to that features mapped using Bing should be tagged with source=Bing, nothing special. Or do you see there any blocker?

They don't request us that it must be possible to track individual changes or objects. It is only mentioned that if something is with source="Ortofotomozaika SR" then it comes from this imagery.

Please let me know if we shold do any other steps but honestly I would really not like to ask them again for something else.

**Comment by andrewharvey** [comment]
> @andrewharvey They actually only adjusted existing text we provided them from wiki. It is a simmilar thing to that features mapped using Bing should be tagged with source=Bing, nothing special. Or do you see there any blocker?

source=Bing is just something that is there to help mappers understand where the change was source, it's not there for legal attribution. You could omit the source=Bing while still tracing from Bing and be okay legally just makes it harder for other mappers to understand where information in the changes were sourced.

Not sure if this is a blocker but just something I noticed and wanted to raise.

**Comment by zdila** [comment]
> source=Bing is just something that is there to help mappers understand where the change was source, it's not there for legal attribution. You could omit the source=Bing while still tracing from Bing and be okay legally just makes it harder for other mappers to understand where information in the changes were sourced.

For source="Ortofotomozaika SR" it sounds the same. Or is it somewhere written something else?

Please let me know what to do to make this pull request merged. Thanks.

**Comment by zdila** [comment]
For completeness let me include my email requestion for explicit Ortofotomozaika SR usage permission:

> Dobrý deň,
> 
> Z [podmienok poskytovania a používania údajov ortofotomozaiky](https://www.geoportal.sk/files/zbgis/objednavky/podmienky_poskytovania_a_pouzivania_udajov_ortofotomozaiky.pdf) komunita OpenStreetMap (ďalej OSM) vyvodila, že dáta je možné použiť pre účely mapovania do OpenStreetMap, ktorá podlieha slobodnej licencii [ODbL](https://opendatacommons.org/licenses/odbl/) 1.0. OSM komunita sa tiež dohodla, že takto odvodené dáta v OSM bude [označovať značkou source=ofmozaika](https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Sources#Ortofotomozaika_SR_.28orthophoto_by_GK.C3.9A_and_NLC.29). Keďže z povahy OSM nie je technicky možné pri každom použití mapy citovať všetky zdroje, odkaz na GKÚ a NLC je spomenutý na [stránke prispievateľov](https://wiki.openstreetmap.org/wiki/Contributors#Slovakia).
> 
> Pre zaradenie Ortofotomozaiky pre účely mapovania do OSM však správcovia registra povolených zdrojov [vyžadujú explicitné povolenie od vlastníka zdroja](https://github.com/osmlab/editor-layer-index/pull/844#issuecomment-673621898), keďže z niektorých pasáži podmienok ku Ortofotomozaike SR vznikli nejasnosti, alebo nejednoznačnosti.
> 
> Preto vás za komunitu OSM prosím o vyjadrenie, či súhlasite s použitím Ortofotomozaiky SR zo stránky https://www.geoportal.sk/sk/udaje/ortofotomozaika/ pre účely mapovania do OSM.
> 
> Prípadne dodatočné otázky vám rád zodpoviem.
> 
> S pozdravom
>

**Comment by zdila** [comment]
Is there still something blocking for this pull request to be merged? Please let me know if there is something I could help with to merge it.

**Comment by grischard** [comment]
Merged, thank you @zdila !

**Comment by zdila** [comment]
Thank you too!

File: sources/europe/sk/OrtofotomozaikaSR.geojson [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "min_zoom": 0,
         "country_code": "SK",
         "type": "tms",
-        "license_url": "https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Sources#Ortofotomozaika_SR_.28orthophoto_by_GK.C3.9A_and_NLC.29"
+        "license_url": "https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Sources#Ortofotomozaika_SR_.28orthophoto_by_GK.C3.9A_and_NLC.29",
+        "privacy_policy_url": "https://wiki.openstreetmap.org/wiki/WikiProject_Slovakia/Sources#Ortofotomozaika_SR_.28orthophoto_by_GK.C3.9A_and_NLC.29"
```
Comment (Klumbumbus) [inline sources/europe/sk/OrtofotomozaikaSR.geojson:16]: How is that a privacy_policy_url?

Comment (zdila) [reply sources/europe/sk/OrtofotomozaikaSR.geojson:16]: What is expected to be in privacy policy? Something from https://www.geoportal.sk/sk/udaje/ortofotomozaika/?

Comment (Klumbumbus) [reply sources/europe/sk/OrtofotomozaikaSR.geojson:16]: Yes, "A URL for the privacy policy of the operator"

Comment (zdila) [reply sources/europe/sk/OrtofotomozaikaSR.geojson:16]: What if there is nothing like that? Or I couldn't find one. Could it be this? https://www.geoportal.sk/files/zbgis/objednavky/podmienky_poskytovania_a_pouzivania_udajov_ortofotomozaiky.pdf

Comment (Klumbumbus) [reply sources/europe/sk/OrtofotomozaikaSR.geojson:16]: No. It must be about the use of personal data (keyword GDPR). Simply remove it if there is none.

Comment (zdila) [reply sources/europe/sk/OrtofotomozaikaSR.geojson:16]: OK, I will remove it then. Thanks.

Comment (zdila) [reply sources/europe/sk/OrtofotomozaikaSR.geojson:16]: https://github.com/osmlab/editor-layer-index/pull/968

**Pull request merged**